### DISCUSSION TITLE
Top-3 Most wanted features

### DISCUSSION THREAD

**Discussion by neSpecc** [codex-team/editor.js#2661] [Announcements]
Write a short list of **the most wanted features** you're missed in Editor.js. Lets find the really most wanted.

<img width="844" alt="top-3-most-wanted" src="https://github.com/codex-team/editor.js/assets/3684889/29581176-5db9-4280-bc0e-4e4f36f03bfa">

Please, be short and stay in a following format:

```
1. Short description (Link to a corresponded discussion if exists)
2. ...
3. ...
```

If you're one of our donators on [OpenCollective](http://opencollective.com/editorjs) or [Patreon](https://www.patreon.com/editorjs), please, include a link to you'r profile. We'll review your siggestion with a higher priority.

### Example
 
1. Blocks drag'n'drop
2. Markdown support
3. Server-side rendering

**Comment by GuillaumeOnepilot** [comment]
1. Integrated History manager (undo/redo feature)
2. JSON output for inline format (would have to migrate everything on my side, but that's a sacrifice I'm willing to make)
3. Native API to handle text selection

**Reply by ats1999** [reply]
I vote for 1 and 2

**Comment by FlusherDock1** [comment]
1. Non-asynchronous way of dumping JSON content in textarea
2. Ability to get raw HTML instead of JSON
3. Option to pin always visible toolbar and block selector
4. Nesting

**Reply by LoopyDev** [reply]
Aww come on...I managed to get nesting nailed down in one weekend ;)

**Reply by Tomsgu** [reply]
> Aww come on...I managed to get nesting nailed down in one weekend ;)

You can make a pull request for a general use if its that trivial.

**Comment by SadEaston** [comment]
1.Support inserting content at the cursor position.
2.Able to input and output HTML with inline styles.
3.Columns Support

**Comment by vigneshksaithal** [comment]
1. Markdown support.
2. Columns support.
3. Nesting.

**Comment by ngmiduc** [comment]
1. Collaborative editing
2. Inline commenting (https://github.com/codex-team/editor.js/discussions/1956)
3. Blocks Drag and Drop

**Comment by tiagozip** [comment]
1. Markdown support
2. Dark mode
3. Inline comments, AI and collaboration

**Comment by javadbat** [comment]
1. ability to choose an " _blank" target link
2. event fire when user delete or remove block
3. support web-component for writing plugin

**Reply by ats1999** [reply]
I vote for 1

**Comment by sethaddison** [comment]
1. Collaborative editing
2. Version history
3. Dark mode

**Comment by ats1999** [comment]
1. Integrated History manager (undo/redo feature)
2. JSON output for inline format (would have to migrate everything on my side, but that's a sacrifice I'm willing to make)

Related Issue -> https://github.com/codex-team/editor.js/issues/2568

**Comment by Tomsgu** [comment]
1. Inline json format
2. Columns tool
3. history

**Comment by tiagozip** [comment]
1. Markdown
2. Markdown
3. Dark mode

**Comment by riskable** [comment]
1. Select text between blocks (not whole blocks) with the ability to delete/cut what's selected.
2. Collaborative editing (really, just the necessary bits to easily integrate with Yjs).
3. Versioning/history with an easy way to keep stuff like that in sync with the server.
